### PR TITLE
Update ECC memory and add SECDED correctors

### PR DESCRIPTION
**Pull request by micprog**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,11 +28,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Expose additional error logging signals
 - Add scrubber to ECC SRAM wrap
 - Add testing signals for tapeout
+- Add secded ECC corrector
```
Comment (LucaRufer) [inline CHANGELOG.md]:
```suggestion
- Add secded ECC corrector
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,11 +28,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Changed
 - Expose  `ecc_sram` ecc error signals
 - Rename cTCLS to ODRG
 - Hide bus ecc behind bender targets, remove related dependencies
+- Update secded testbench to use correctors and fix error injection
```
Comment (LucaRufer) [inline CHANGELOG.md]:
```suggestion
- Update secded testbench to use correctors and fix error injection
```

File: CHANGELOG.md (hunk 1 above)
Comment (LucaRufer) [reply CHANGELOG.md]: These changes should be listed under "Unreleased" and not under 0.5.1.

**Review by LucaRufer — CHANGES_REQUESTED** [review]
The changes look good to me. I focused my review on the ECC encode/correct modules and the corresponding testbench. The only comment I have is a mistake I made in the changelog file.

**Review by yvantor — APPROVED** [review]
Looks good to me!

**Pull request merged**